tests/foreman/api/test_convert2rhel.py

```diff
@@ -0,0 +1,314 @@
+    repo.product = product
+    repo.sync()
+    return repo
+
+
+def create_activation_key(org, lce, cv, subscription_id):
+    """Create activation key with subscription"""
+    act_key = entities.ActivationKey(
+        organization=org,
+        content_view=cv,
+        environment=lce,
+    ).create()
+    act_key.add_subscriptions(data={'subscription_id': subscription_id})
+    return act_key
+
+
+def update_cv(cv, lce, repos):
+    """Update and publish Content view with repos"""
+    cv = entities.ContentView(id=cv.id, repository=repos).update(["repository"])
+    cv.publish()
+    cv = cv.read()
+    promote(cv.version[-1], environment_id=lce.id)
+    return cv
+
+
+def register_host(act_key, module_org, module_loc, hostname, ubi=None):
+    """Register host to satellite"""
+    # generate registration command
+    command = entities.RegistrationCommand(
+        organization=module_org,
+        activation_keys=[act_key.name],
+        location=module_loc,
+        insecure=True,
+        repo=ubi,
+    ).create()['registration_command']
+    result = ssh.command(command, hostname=hostname)
+    assert result.status == 0
+
+
+@pytest.fixture(scope="module")
+def ssl_cert(module_org):
+    """Create credetial with SSL cert for Oracle Linux"""
+    res = requests.get(settings.repos.convert2rhel.ssl_cert_oracle)
+    res.raise_for_status()
+    return entities.ContentCredential(
+        content=res.text, organization=module_org, content_type='cert'
+    ).create()
+
+
+@pytest.fixture(scope='module')
+def manifest(module_org):
+    with manifests.clone() as manifest:
+        upload_manifest(module_org.id, manifest.content)
+
+
+@pytest.fixture
+def activation_key_rhel(module_org, module_lce, module_promoted_cv, version):
+    """Create activation key that will be used after conversion for registration"""
+    subs = entities.Subscription(organization=module_org).search(
+        query={'search': f'{DEFAULT_SUBSCRIPTION_NAME}'}
+    )
+    assert subs
+    return create_activation_key(module_org, module_lce, module_promoted_cv, subs[0].id)
+
+
+@pytest.fixture(scope='module')
+def enable_rhel_subscriptions(module_org, manifest, version):
+    """Enable and sync RHEL rpms repos"""
+    major = version.split('.')[0]
+    minor = ""
+    if major == '8':
+        repo_names = ['rhel8_bos', 'rhel8_aps']
+        minor = version[1:]
+    else:
+        repo_names = ['rhel7']
+
+    rh_repos = []
+    for name in repo_names:
+        rh_repo_id = enable_rhrepo_and_fetchid(
+            basearch=DEFAULT_ARCHITECTURE,
+            org_id=module_org.id,
+            product=REPOS[name]['product'],
+            repo=REPOS[name]['name'] + minor,
+            reposet=REPOS[name]['reposet'],
+            releasever=REPOS[name]['releasever'] + minor,
+        )
+        # Sync step because repo is not synced by default
+        rh_repo = entities.Repository(id=rh_repo_id).read()
+        rh_repo.sync(timeout=2500)
+        rh_repos.append(rh_repo)
```

Comment:
I can definitely make the first repo synchronization async, but I would keep the second one synchronous. To make sure all repos are synced before starting the conversion.

Reply:
I see that you have made the changes already. If the first async sync would fail, you would get to know it once you want to consume its content. Isn't it too late? Would implementing some wait_for logic make sense here?

Reply:
:+1: adding wait_for logic and all syncs now can be async with assert on task status on each sync